tests/templates/test_state_preparations/test_state_prep_mps.py

```diff
@@ -19,234 +19,583 @@
+        dev = qml.device("default.qubit")
+
+        qs = qml.tape.QuantumScript(
+            qml.MPSPrep.compute_decomposition(
+                mps, wires=wires["state"], work_wires=wires["work"], right_canonicalize=True
+            ),
+            [qml.state()],
+        )
+        output = dev.execute(qs)[: 2**num_wires]
+
+        assert np.allclose(state, output, rtol=0.01)
+
+    def test_decomposition(self):
+        """Tests that the template defines the correct decomposition."""
+
+        mps = [
+            np.array([[0.70710678, 0.0], [0.0, 0.70710678]]),
+            np.array(
+                [
+                    [[0.0, 1.0, 0.0, 0.0], [0.0, 0.0, 0.0, 0.0]],
+                    [[0.0, 0.0, -0.0, 0.0], [-1.0, 0.0, 0.0, 0.0]],
+                ]
+            ),
+            np.array(
+                [
+                    [[0.00000000e00, 1.74315280e-32], [-7.07106781e-01, -7.07106781e-01]],
+                    [[7.07106781e-01, 7.07106781e-01], [0.00000000e00, 0.00000000e00]],
+                    [[0.00000000e00, 0.00000000e00], [-7.07106781e-01, 7.07106781e-01]],
+                    [[-7.07106781e-01, 7.07106781e-01], [0.00000000e00, 0.00000000e00]],
+                ]
+            ),
+            np.array([[1.0, 0.0], [0.0, 1.0]]),
+        ]
+
+        ops = qml.MPSPrep.compute_decomposition(mps, wires=range(2, 6), work_wires=[0, 1])
+
+        for ind, op in enumerate(ops):
+            assert op.wires == qml.wires.Wires([2 + ind] + [0, 1])
+            assert op.name == "QubitUnitary"
+
+    @pytest.mark.parametrize(
+        ("work_wires", "msg"),
+        [(None, "The qml.MPSPrep decomposition requires"), (1, "The bond dimension cannot exceed")],
+    )
+    def test_wires_decomposition(self, work_wires, msg):
+        """Checks that error is shown if no `work_wires` are given in decomposition"""
+
+        mps = [
+            np.array([[0.70710678, 0.0], [0.0, 0.70710678]]),
+            np.array(
+                [
+                    [[0.0, 1.0, 0.0, 0.0], [0.0, 0.0, 0.0, 0.0]],
+                    [[0.0, 0.0, -0.0, 0.0], [-1.0, 0.0, 0.0, 0.0]],
+                ]
+            ),
+            np.array(
+                [
+                    [[0.00000000e00, 1.74315280e-32], [-7.07106781e-01, -7.07106781e-01]],
+                    [[7.07106781e-01, 7.07106781e-01], [0.00000000e00, 0.00000000e00]],
+                    [[0.00000000e00, 0.00000000e00], [-7.07106781e-01, 7.07106781e-01]],
+                    [[-7.07106781e-01, 7.07106781e-01], [0.00000000e00, 0.00000000e00]],
+                ]
+            ),
+            np.array([[1.0, 0.0], [0.0, 1.0]]),
+        ]
+
+        op = qml.MPSPrep(mps, wires=range(2, 5), work_wires=work_wires)
+        with pytest.raises(ValueError, match=msg):
+            op.decomposition()
+
+    def test_right_canonical(self):
+        """Checks that the function `_mps_to_right_canonical_representation` generates mps in the correct format"""
```

Comment:
```suggestion
        """Checks that the function `right_canonicalize_mps` generates mps in the correct format"""
```

Reply:
Might want to add a testcase to validate that the `right_canonicalize` keyword works for sure.

Reply:
In the correctness tests I added a non canonicalize example 😄